### PR TITLE
New docs UI with Table of Contents

### PR DESCRIPTION
**Pull request by microsoftsam**
screen shot 2017-03-10 at 5 58 43 pm
screen shot 2017-03-10 at 5 58 37 pm
screen shot 2017-03-10 at 5 58 30 pm
screen shot 2017-03-10 at 5 58 22 pm
screen shot 2017-03-10 at 5 58 10 pm
screen shot 2017-03-10 at 5 58 06 pm

Table of contents is authored in SUMMARY.md file under docs, like so:

```md
# Summary

### Projects

* [Projects](/projects.md)
    * [Games]()
        * [Flashing Heart](/projects/flashing-heart.md)
        * [Smiley Buttons](/projects/smiley-buttons.md)
        * [Love Meter](/projects/love-meter.md)
        * [Rock Paper Scissors](/projects/rock-paper-scissors.md)
        * [Magic Button Trick](/projects/magic-button-trick.md)
    * [Music]()
        * [Hack your headphones](/projects/hack-your-headphones.md)
        * [Banana Keyboard](/projects/banana-keyboard.md)
        * [Guitar](/projects/guitar.md)
    * [Fashion]()
        * [Duct Tape Wallet](/projects/wallet.md)
        * [Watch](/projects/watch.md)
    * [Toys]()
        * [Inchworm](/projects/inchworm.md)
    * [More]()
        * [Timing Gates](/projects/timing-gates.md)
        * [Compass](/projects/compass.md)
        * [Telegraph](/projects/telegraph.md)
        * [Radio](/projects/radio.md)

```

### CODE REVIEW

**Comment by pelikhan** [comment]
You can drop the .md in the generated links.

**Comment by microsoftsam** [comment]
use pxt-microbit's branch: **newdocs** to test

**Comment by pelikhan** [comment]
Breadcrumb color should be white
image

**Comment by pelikhan** [comment]
- [ ] project categories like Game show up as links and go nowhere.

**Comment by pelikhan** [comment]
Format seems to have duplication. For example there are 2 "projects" title...

```
### Projects

* [Projects](/projects)
```

**Comment by microsoftsam** [comment]
project categories showing up with no links is due to the authoring: 
```
    * [Game]()
```
Fixed that by removing that option (to add categories with no links).

No need to duplicate Projects. (The first one is just a comment). I was hoping to use those as heading, and maybe in the future we can. but they're not required.

**Comment by microsoftsam** [comment]
Added Bing site search: 
screen shot 2017-03-11 at 6 48 40 pm

File: pxtlib/docsrender.ts [hunk 1]
```diff
@@ -172,6 +175,31 @@ namespace pxt.docs {
...
         }
 
+        let currentTocEntry: TOCMenuEntry;
+        let recTOC = (m: TOCMenuEntry, lev: number) => {
+            let templ = toc["item"]
+            let mparams: Map<string> = {
+                NAME: m.name,
+            }
+            if (m.path && !/^(https?:|\/)/.test(m.path))
+                return error("Invalid link: " + m.path)
```
Comment (pelikhan) [inline pxtlib/docsrender.ts]: The foundation is an https link

Comment (microsoftsam) [inline pxtlib/docsrender.ts]: Same logic as previous docs menu, see recDoc above. How is this any different?

**Pull request merged**